src/HotChocolate/Fusion/src/Core/Planning/Pipeline/ExecutionStepDiscoveryMiddleware.cs

```diff
@@ -797,9 +797,9 @@ private bool EnsureStepCanBeResolvedFromRoot(
 
         while (current is not null)
         {
-            var typeMetadata = _config.GetType<ObjectTypeMetadata>(path.Selection.DeclaringType.Name);
+            var typeMetadata = _config.GetType<ObjectTypeMetadata>(current.Selection.DeclaringType.Name);
 
-            if (!typeMetadata.Fields[path.Selection.Field.Name].Bindings.ContainsSubgraph(subgraphName))
+            if (!typeMetadata.Fields[current.Selection.Field.Name].Bindings.ContainsSubgraph(subgraphName))
```

Comment:
This ensures that we fail the query plan generation early instead of either failing later or producing an invalid query plan that fails at runtime.